Graphics/ShaderTools/src/ShaderToolsCommon.cpp

```diff
@@ -327,4 +331,32 @@ bool ProcessShaderIncludes(const ShaderCreateInfo& ShaderCI, std::function<void(
     }
 }
 
+void MergeShaderIncludes(const ShaderCreateInfo& ShaderCI, IDataBlob** ppData)
+{
+    VERIFY_EXPR(ShaderCI.Desc.Name != nullptr);
+    VERIFY_EXPR(ppData != nullptr);
+
+    std::stringstream Stream;
+
+    auto IncludeHandler = [&](IDataBlob* pData, const Char* Path) {
+        Stream << String{static_cast<const Char*>(pData->GetConstDataPtr()), pData->GetSize()} << std::endl;
```

Comment:
I think you can use `Stream.write` here to avoid creating temporary string and allocating memory that will be released immediately after